docs/Reference/CLI/CLI-Subcommands.md

```diff
@@ -59,3 +59,314 @@ Generate a list of peer IDs including the private key, public key, and peer ID w
 * `--outputFile` is the path and filename of the file to which the keys are written. If not specified, the peer IDs
 are written to the `./config/peer-ids.dat` file. 
 * `<number` is the number of peer IDs to generate.
+
+## validator
+
+Register or generate validators and send transactions to an Ethereum 1.0 node.
+
+### generate
+
+Register a validator by generating new keys and sending deposit transactions to an Ethereum 1.0
+node.
+
+#### deposit-amount-gwei
+
+```bash tab="Syntax"
+teku validator generate --deposit-amount-gwei=<GWEI>
+```
+
+```bash tab="Example"
+teku validator generate --deposit-amount-gwei=32000000000
+```
+
+Amount to deposit in the Ethereum 1.0 deposit contract.
+
+#### encrypted-keystore-enabled
+
+```bash tab="Syntax"
+teku validator generate --encrypted-keystore-enabled=<BOOLEAN>
+```
+
+```bash tab="Example"
+teku validator generate --encrypted-keystore-enabled=false
+```
+
+Specify whether to create encrypted V3 keystore validator and withdrawal keys.
+Defaults to `true`.
```

Comment:
Outputting keys directly to the command line is very useful when you're running this via docker.  Writing to a file just outputs to the docker file system and they are then deleted when the command exits and the container shuts down.

As I understand it there are three modes:
1. Encrypted to file
2. Unencrypted to file
3. Unencrypted to stdout.

1 and 3 are extremely useful.  2 is handy for compatibility because not all tools support encrypted keystores but the unencrypted yaml file is pretty commonly accepted.

Reply:
@jakehaugen @ajsutton as per Jake's suggestion, for option 3, we are also printing the "warning". In our tests, we use developer option to suppress that (we also suppress the warning if stdout redirection is used). If this warning needs to go away entirely we may need a ticket to be raised to address that.

Reply:
I think it's reasonable to print a warning when output is to the console - just need to not print it when output is being redirected.  We should test that running in docker and redirecting the output doesn't print the warning.